zetaclient/chains/evm/observer/v2_inbound.go

```diff
@@ -410,16 +412,16 @@ func (ob *Observer) parseAndValidateDepositAndCallEvents(
 	// collect and sort events by block number, then tx index, then log index (ascending)
 	events := make([]*gatewayevm.GatewayEVMDepositedAndCalled, 0)
 	for iterator.Next() {
-		events = append(events, iterator.Event)
 		err := common.ValidateEvmTxLog(&iterator.Event.Raw, gatewayAddr, "", common.TopicsGatewayDepositAndCall)
 		if err == nil {
 			events = append(events, iterator.Event)
 			continue
 		}
-		ob.Logger().Inbound.Warn().
-			Err(err).
-			Msgf("ObserveGateway: invalid DepositedAndCalled event in tx %s on chain %d at height %d",
-				iterator.Event.Raw.TxHash.Hex(), ob.Chain().ChainId, iterator.Event.Raw.BlockNumber)
+		ob.Logger().
+			Inbound.Warn().
+			Stringer(logs.FieldTx, iterator.Event.Raw.TxHash).
+			Uint64(logs.FieldBlock, iterator.Event.Raw.BlockNumber).
+			Msg("invalid DepositedAndCalled event")
 	}
```

Comment:
A comment before the sort function would be good

Reply:
https://github.com/zeta-chain/node/pull/3517/commits/e07d5288abb5dbecc3c86964d0407119208b2b05